### PR TITLE
fix: use Mojang log4j config file fixes by version

### PR DESCRIPTION
**Pull request by itzg**
Part of #1177

### CODE REVIEW

**Review by StoneLabs — COMMENTED** [review]
looks good to me at first sight

**Comment by itzg** [comment]
Verified with `-e SETUP_ONLY=true` and

Version | Runner line
--------|----------------
1.12.2 | java -Xmx1G -Xms1G -Dlog4j.configurationFile=log4j2_112-116.xml -jar minecraft_server.1.12.2.jar
1.11.2 | java -Xmx1G -Xms1G -Dlog4j.configurationFile=log4j2_17-111.xml -jar minecraft_server.1.11.2.jar
1.18 | java -Xmx1G -Xms1G -Dlog4j2.formatMsgNoLookups=true -jar minecraft_server.1.18.jar

**Comment by StoneLabs** [comment]
wait, we dont need -Dlog4j2.formatMsgNoLookups=true for 1.18.1 instances

**Comment by itzg** [comment]
> wait, we dont need `-Dlog4j2.formatMsgNoLookups=true for 1.18.1` instances

Ah yes; however, I originally thought it couldn't hurt.

File: scripts/start-finalExec [hunk 1]
```diff
@@ -85,8 +85,24 @@ if [ -n "$JVM_DD_OPTS" ]; then
...
+  JVM_OPTS="-Dlog4j.configurationFile=${file} ${JVM_OPTS}"
+}
+
 # Patch Log4j remote code execution vulnerability
-JVM_OPTS="-Dlog4j2.formatMsgNoLookups=true ${JVM_OPTS}"
+if [[ ${TYPE^^} != PAPER ]] && versionLessThan 1.12; then
+  patchLog4jConfig log4j2_17-111.xml https://launcher.mojang.com/v1/objects/dd2b723346a8dcd48e7f4d245f6bf09e98db9696/log4j2_17-111.xml
+elif [[ ${TYPE^^} != PAPER ]] && versionLessThan 1.17; then
+  patchLog4jConfig log4j2_112-116.xml https://launcher.mojang.com/v1/objects/02937d122c86ce73319ef9975b58896fc1b491d1/log4j2_112-116.xml
+else
```
Comment (StoneLabs) [inline scripts/start-finalExec:103]: `elif  [[ ${TYPE^^} != PAPER ]] && versionLessThan 1.18.1; then`

should be fine because 1.18.1 is safe to use

Comment (itzg) [reply scripts/start-finalExec]: Yep just pushed a change to that. Please re-review.

Comment (itzg) [reply scripts/start-finalExec]: oops, 1.18.1 not 1.18...one more push coming

File: scripts/start-finalExec [hunk 2]
```diff
@@ -96,11 +96,12 @@ patchLog4jConfig() {
...
 # Patch Log4j remote code execution vulnerability
-if [[ ${TYPE^^} != PAPER ]] && versionLessThan 1.12; then
+# NOTE: Paper, bukkit, and spigot are normalized to SPIGOT
+if [[ ${TYPE^^} != SPIGOT ]] && versionLessThan 1.12; then
   patchLog4jConfig log4j2_17-111.xml https://launcher.mojang.com/v1/objects/dd2b723346a8dcd48e7f4d245f6bf09e98db9696/log4j2_17-111.xml
-elif [[ ${TYPE^^} != PAPER ]] && versionLessThan 1.17; then
+elif [[ ${TYPE^^} != SPIGOT ]] && versionLessThan 1.17; then
   patchLog4jConfig log4j2_112-116.xml https://launcher.mojang.com/v1/objects/02937d122c86ce73319ef9975b58896fc1b491d1/log4j2_112-116.xml
-else
+elif versionLessThan 1.18; then
```
Comment (StoneLabs) [inline scripts/start-finalExec]: I do believe 1.18 requires the fix. Only 1.18.1 is not affected.
image

Comment (itzg) [reply scripts/start-finalExec]: Yep, caught that midflight while reading your previous comment.

**Review by StoneLabs — COMMENTED** [review]
I think we're good now (as long as the tests pass and you verified by sample using SETUP_ONLY)

There might be some edge cases with versions and types where its not really needed. But it should be fine.

**Comment by Stealthii** [comment]
Can confirm changes for a variety of non-Spigot servers.

Is there a reason we're not applying the configuration mitigation for Spigot-like servers? Not all variations (Airplane, etc.) necessarily have log4j patches in place.

The only other change I'd recommend is to roll off the config patch pre 1.7 (as 1.7 and below are [marked as not vulnerable](https://www.minecraft.net/sv-se/article/important-message--security-vulnerability-java-edition), and this option might not work on older versions, such as Canyon)

**Comment by StoneLabs** [comment]
> The only other change I'd recommend is to roll off the config patch pre 1.7 (as 1.7 and below are [marked as not vulnerable](https://www.minecraft.net/sv-se/article/important-message--security-vulnerability-java-edition), and this option might not work on older versions, such as Canyon)

I don't know how I missed that. You are right of course.
This should be a quick change though and I'm sure @itzg will do it soon.
